src/framework/core/js/FluidViewDebugging.js

```diff
@@ -0,0 +1,649 @@
+            };
+        });
+    };
+
+    fluid.debug.domIdtoHighlightId = function (domId) {
+        return "highlight-for:" + domId;
+    };
+    
+    fluid.debug.highlighter.highlight = function (that, highlightRoot, dispositions) {
+        var p = fluid.debug.highlighter.positionProps;
+        for (var i = 0; i < dispositions.length; ++ i) {
+            var disp = dispositions[i];
+            var container = disp.container;
+            if (disp.noHighlight) {
+                continue;
+            }
+ 
+            var highlight = $(that.options.markup.highlightElement);
+            highlight.prop("id", fluid.debug.domIdtoHighlightId(container.prop("id")));
+            highlightRoot.append(highlight);
+                
+            highlight.css("background-color", fluid.debug.arrayToRGBA(disp.colour));
+            for (var j = 0; j < p.length; ++ j) {
+                highlight.css(p[j], container.css(p[j] || ""));
+            }
+            var offset = container.offset();
+            var containerBody = container[0].ownerDocument.body;
+            if (containerBody !== document.body) { // TODO: This primitive algorithm will not account for nested iframes
+                offset.left -= $(containerBody).scrollLeft();
+                offset.top -= $(containerBody).scrollTop();
+            }
+            highlight.offset(offset);
+            
+        }
+    };
+    
+    fluid.debug.ignorableGrades = ["fluid.debug.listeningView", "fluid.debug.listeningPanel", "fluid.debug.listeningRenderer"];
+    
+    fluid.debug.frameworkGrades = fluid.frameworkGrades;
+    
+    fluid.debug.filterGrades = function (gradeNames) {
+        var highestFrameworkIndex = -1;
+        var output = [];
+        fluid.each(gradeNames, function (gradeName) { // TODO: remove fluid.indexOf
+            var findex = fluid.debug.frameworkGrades.indexOf(gradeName);
+            if (findex > highestFrameworkIndex) {
+                highestFrameworkIndex = findex;
+            } else if (findex === -1 && fluid.debug.ignorableGrades.indexOf(gradeName) === -1 && gradeName.indexOf("{") === -1) {
+                output.push(gradeName);
+            }
+        });
+        output.push(fluid.debug.frameworkGrades[highestFrameworkIndex]);
+        return output;
+    };
+    
+    fluid.debug.renderDefaults = function (typeName, options) {
+        var string = "fluid.defaults(\"" + typeName + "\", " + JSON.stringify(options, null, 4) + ");\n";
+        var markup = "<pre>" + string + "</pre>";
+
+        return markup;
+    };
+    
+    fluid.debug.renderSelectorUsageRecurse = function (source, segs, options) {
+        if (fluid.isPrimitive(source)) {
+            if (typeof(source) === "string" && source.indexOf(options.findString) !== -1) {
+                var path = segs.slice(0, 2);
+                fluid.set(options.target, path, fluid.copy(fluid.get(options.fullSource, path)));
```

Comment:
Let's assign the value of fluid.get() to a variable before copying and setting it.